mkt/webapps/decorators.py

```diff
@@ -39,26 +38,8 @@ def wrapper(request, addon_id, addon, *args, **kw):
 
 def app_view(f, qs=Webapp.objects.all):
     @functools.wraps(f)
-    def wrapper(request, addon_id=None, app_slug=None, *args,
-                **kw):
-        """Provides an addon given either an addon_id or app_slug."""
-        assert addon_id or app_slug, 'Must provide addon_id or app_slug'
-
-        def get(**kw):
-            return get_object_or_404(qs(), **kw)
-
-        if addon_id and addon_id.isdigit():
-            addon = get(id=addon_id)
-            # Don't get in an infinite loop if addon.app_slug.isdigit().
-            if addon.app_slug != addon_id:
-                url = request.path.replace(addon_id, addon.app_slug, 1)
-                if request.GET:
-                    url += '?' + request.GET.urlencode()
-                return http.HttpResponsePermanentRedirect(url)
-        elif addon_id:
-            addon = get(app_slug=addon_id)
-        elif app_slug:
-            addon = get(app_slug=app_slug)
```

Comment:
All of this was unused - addon_id was only passed for add-on pages, all app views passed app_slug instead.